src/hypothesis/strategies.py

```diff
@@ -695,6 +695,32 @@ def text(
 
 @cacheable
 @defines_strategy
+def strings_matching_regex(regex):
+    """Return strategy that generates strings that match given regex.
+
+    Regex can be either a string or compiled regex (through
+    :func:`re.compile`).
```

Comment:
To clarify that it's unicode-only, now that the name has changed to make sense in future when it isn't:

    ``regex`` can be either a unicode string or compiled unicode regex (with :func:`python:re.compile`).  Byte-string support may be added in a future version.